classes/opponentPhaseBullethell3.lua

```diff
@@ -0,0 +1,101 @@
+-- The template opponent phase
+
+local opponentPhaseBullethell3 = {}
+opponentPhaseBullethell3.__index = opponentPhaseBullethell3
+
+
+local ORBIT_START_ANG = -math.pi / 2
+local ORBIT_ANG_SPEED = 1.2 	-- How many radians per second should the opponent advance in orbit?
+local ORBIT_RADIUS_X = 400
+local ORBIT_RADIUS_Y = 250
+
+local MAX_TURN_SPEED = 1
+
+local ORBIT_FOCAL_X = ARENA_WIDTH / 2
+local ORBIT_FOCAL_Y = ARENA_HEIGHT / 2
+
+opponentPhaseBullethell3.RADIUS = 55
+opponentPhaseBullethell3.SPAWN_X = ORBIT_FOCAL_X + ORBIT_RADIUS_X * math.cos(ORBIT_START_ANG)
+opponentPhaseBullethell3.SPAWN_Y = ORBIT_FOCAL_Y + ORBIT_RADIUS_Y * math.sin(2 * ORBIT_START_ANG)
+opponentPhaseBullethell3.NUM_LIVES = 12
+
+
+--------------------
+-- MAIN CALLBACKS --
+--------------------
+
+function opponentPhaseBullethell3.new()
+	local self = classes.opponentBase.new(opponentPhaseBullethell3)
+
+	self.orbitAng = ORBIT_START_ANG
+	if state.player then
+		self.angle = math.atan2(state.player.y - self.y, state.player.x - self.x)
+	end
+	self.shotState = "focus"
+
+	self.currentCooldown = 0
+	self.bulletCooldownTime = 1
+	self.numBullets = 3
+
+	return self
+end
+
+function opponentPhaseBullethell3:update(dt)
+	-- Call superclass method
+	classes.opponentBase.update(self, dt)
+
+	if self.stunned then return end
+
+	--Aim
+	if state.player then
+		self.angle = math.atan2(state.player.y - self.y, state.player.x - self.x)
+	end
```

Comment:
There should probably be a turn speed limit here rather than snapping perfectly to the player